presto-testing-docker/src/main/java/io/prestosql/testing/docker/DockerContainer.java

```diff
@@ -56,27 +55,30 @@
     private static final boolean DEBUG = false;
 
     private static final String HOST_IP = "127.0.0.1";
-    private final String image;
-    private final Map<String, String> environment;
+
     private DockerClient dockerClient;
     private String containerId;
 
     private Map<Integer, Integer> hostPorts;
 
     public DockerContainer(String image, List<Integer> ports, Map<String, String> environment, List<String> capabilities, CheckedConsumer<HostPortProvider> healthCheck)
     {
-        this.image = requireNonNull(image, "image is null");
-        this.environment = ImmutableMap.copyOf(requireNonNull(environment, "environment is null"));
+        requireNonNull(image, "image is null");
+        requireNonNull(ports, "ports is null");
+        requireNonNull(environment, "environment is null");
+        requireNonNull(capabilities, "capabilities is null");
+        requireNonNull(healthCheck, "healthCheck is null");
+
         try {
-            startContainer(ports, healthCheck, capabilities);
+            startContainer(image, ports, environment, capabilities, healthCheck);
         }
         catch (Exception e) {
             closeAllSuppress(e, this);
             throw new RuntimeException(e);
         }
     }
 
-    private void startContainer(List<Integer> ports, CheckedConsumer<HostPortProvider> healthCheck, List<String> capabilities)
+    private void startContainer(String image, List<Integer> ports, Map<String, String> environment, List<String> capabilities, CheckedConsumer<HostPortProvider> healthCheck)
```

Comment:
I'd prefer to go into opposite direction. `DockerContainer` is stateful anyway and it doesn't help it if we pass configuration explicitly to methods.